### PR TITLE
Add tests and validation to bpp

### PR DESCRIPTION
**Pull request by ajax-yurkovskyi-t**
Add tests and validation for negative values to bpp

### CODE REVIEW

File: src/main/kotlin/com/example/iot_management_device/beanpostprocessor/MethodAttempt.kt [hunk 1]
```diff
@@ -26,6 +27,6 @@ data class MethodAttempt(
 
     fun getRemainingLockoutTime(): Long {
         val remainingTime = lockoutEndTimeMillis.get() - System.currentTimeMillis()
-        return (remainingTime / 1000)
+        return (remainingTime / TimeUnit.SECONDS.toMillis(1))
```
Comment (ajax-shulepa-a) [inline src/main/kotlin/com/example/iot_management_device/beanpostprocessor/MethodAttempt.kt:30]:
```suggestion
        return TimeUnit.MILLISECONDS.toSeconds(remainingTime)
```

File: src/main/kotlin/com/example/iot_management_device/beanpostprocessor/MethodAttemptLimiter.kt [hunk 2]
```diff
@@ -1,8 +1,11 @@
...
+import com.example.iot_management_device.beanpostprocessor.MethodAttemptLimits.DEFAULT_LOCKOUT_DURATION_MILLIS
+import com.example.iot_management_device.beanpostprocessor.MethodAttemptLimits.DEFAULT_MAX_ATTEMPTS
+
 @Retention(AnnotationRetention.RUNTIME)
 @Target(AnnotationTarget.FUNCTION)
 annotation class MethodAttemptLimiter(
-    val maxAttempts: Int = 5,
-    val lockoutDurationMillis: Long = 5 * 60 * 1000, // 5 minutes in milliseconds
+    val maxAttempts: Int = DEFAULT_MAX_ATTEMPTS,
+    val lockoutDurationMillis: Long = DEFAULT_LOCKOUT_DURATION_MILLIS, // 5 minutes in milliseconds
```
Comment (ajax-shulepa-a) [inline src/main/kotlin/com/example/iot_management_device/beanpostprocessor/MethodAttemptLimiter.kt:10]: move comment to constant declaration

File: src/main/kotlin/com/example/iot_management_device/beanpostprocessor/MethodAttemptLimiterBeanPostProcessor.kt [hunk 3]
```diff
@@ -55,13 +57,27 @@ class MethodAttemptLimiterBeanPostProcessor : BeanPostProcessor {
             attempt.incrementAttempts()
             if (attempt.hasExceededLimit()) {
                 attempt.lockOut()
-                throw AttemptLimitReachedException(ATTEMPT_LIMIT_MESSAGE.format(methodName, attempt.lockoutDurationMillis / 1000))
+                throw AttemptLimitReachedException(
+                    ATTEMPT_LIMIT_MESSAGE.format(
+                        methodName,
+                        attempt.lockoutDurationMillis / TimeUnit.SECONDS.toMillis(1)
```
Comment (ajax-shulepa-a) [inline src/main/kotlin/com/example/iot_management_device/beanpostprocessor/MethodAttemptLimiterBeanPostProcessor.kt:63]: simplify

**Review by ajax-shulepa-a — APPROVED** [review]
